// Copyright 2022 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfa_virtual_people;

message BinaryParameter {
  string name = 1;

  string value = 2;

  optional string golden = 3;

  optional string proto = 4;

  repeated string proto_dependencies = 5;

  optional string proto_type = 6;
}

message TestCase {
  string name = 1;

  repeated BinaryParameter binary_parameters = 2;
} 

message IntegrationTest {
  string name = 1;

  string binary = 2;

  repeated TestCase test_cases = 3;
}

message IntegrationTestList {
  string workspace_name = 1;

  repeated IntegrationTest tests = 2;
}
